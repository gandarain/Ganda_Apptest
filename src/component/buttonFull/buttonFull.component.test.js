import React from 'react'
import renderer from 'react-test-renderer'

import ButtonFull from './buttonFull.component'

describe('ButtonFull Component', () => {
  const mockProps = {
    title: 'title',
    onPress: jest.fn(),
    withBorder: false
  }

  it('should render button full without border', () => {
    const rendered = renderer.create(<ButtonFull {...mockProps} />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render button full with border', () => {
    mockProps.withBorder = true
    const rendered = renderer.create(<ButtonFull {...mockProps} />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
