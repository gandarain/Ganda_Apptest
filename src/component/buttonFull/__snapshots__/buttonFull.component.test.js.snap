// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonFull Component should render button full with border 1`] = `
<View
  accessibilityLabel="ButtonFull"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#FF6049",
      "borderRadius": 8,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 15,
      "width": "100%",
    }
  }
>
  <Text
    accessibilityLabel="TextButtonFull"
    style={
      {
        "color": "#FF6049",
        "fontFamily": "Roboto-Bold",
        "fontSize": 20,
      }
    }
  >
    title
  </Text>
</View>
`;

exports[`ButtonFull Component should render button full without border 1`] = `
<View
  accessibilityLabel="ButtonFull"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FF6049",
      "borderColor": "#FF6049",
      "borderRadius": 8,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 15,
      "width": "100%",
    }
  }
>
  <Text
    accessibilityLabel="TextButtonFull"
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Roboto-Bold",
        "fontSize": 20,
      }
    }
  >
    title
  </Text>
</View>
`;
